diff --git a/src/video/uikit/SDL_uikitviewcontroller.m b/src/video/uikit/SDL_uikitviewcontroller.m
index 49a39b6bc..81ad6820b 100644
--- a/src/video/uikit/SDL_uikitviewcontroller.m
+++ b/src/video/uikit/SDL_uikitviewcontroller.m
@@ -401,6 +401,11 @@ - (void)textFieldTextDidChange:(NSNotification *)notification
             SDL_SendKeyboardText([changeText UTF8String]);
         }
         changeText = nil;
+    } else if (changeText!=nil && textField.markedTextRange != nil){
+        NSUInteger len = changeText.length;
+        if (len > 0) {
+            SDL_SendEditingText([[[textField textInRange: textField.markedTextRange] stringByReplacingOccurrencesOfString:@"\u2006" withString:@""] UTF8String], 0, 0);
+        }
     }
 }

@@ -448,6 +453,8 @@ - (BOOL)textField:(UITextField *)_textField shouldChangeCharactersInRange:(NSRan
             /* it wants to replace text with nothing, ie a delete */
             SDL_SendKeyboardKey(SDL_PRESSED, SDL_SCANCODE_BACKSPACE);
             SDL_SendKeyboardKey(SDL_RELEASED, SDL_SCANCODE_BACKSPACE);
+        } else {
+            SDL_SendEditingText([[[textField textInRange: textField.markedTextRange] stringByReplacingOccurrencesOfString:@"\u2006" withString:@""] UTF8String], 0, 0);
         }
         if (textField.text.length < 16) {
             textField.text = obligateForBackspace;
